src/deluge/model/instrument/midi_instrument.cpp

```diff
@@ -212,25 +220,39 @@ int32_t MIDIInstrument::getOutputMasterChannel() {
 }
 
 void MIDIInstrument::monophonicExpressionEvent(int32_t newValue, int32_t whichExpressionDimension) {
+	lastMonoExpression[whichExpressionDimension] = newValue;
+	sendMonophonicExpressionEvent(whichExpressionDimension);
+}
+
+void MIDIInstrument::sendMonophonicExpressionEvent(int32_t whichExpressionDimension) {
 
 	int32_t masterChannel = getOutputMasterChannel();
 
 	switch (whichExpressionDimension) {
 	case 0: {
+		int32_t newValue = add_saturation(lastCombinedPolyExpression[whichExpressionDimension],
+		                                  lastMonoExpression[whichExpressionDimension]);
 		int32_t valueSmall = (newValue >> 18) + 8192;
 		midiEngine.sendPitchBend(masterChannel, valueSmall & 127, valueSmall >> 7, channel);
 		break;
 	}
 
-	case 1:
+	case 1: {
+		//mono y expression is limited to positive values, double it to match range
+		int32_t polyPart = (lastCombinedPolyExpression[whichExpressionDimension] >> 25) + 64;
+		int32_t monoPart = lastMonoExpression[whichExpressionDimension] >> 24;
```

Comment:
Should we sum these in a higher precision fixed-point space (i.e. prior to the shifts) to get better rounding? I doubt it matters much in practice.

Reply:
They both fully max out the int32 range so they need to shift down first anyway, this way it's easier to handle the fact that one is bipolar and the other is unipolar